[licongshen12] iP

licongshen12 opened the pull request:
# ~~Duke~~ Davidson
> "Productivity is never an accident. It is always the result of a commitment to excellence, intelligent planning, and focused effort." — Paul J. Meyer
## Davidson provides comprehensive support for task management. It is:
- easy to learn
- convenient
- and now comes with an interactive GUI
## All you need to do is,
1. Download it from [here](https://github.com/licongshen12/ip/releases/tag/A-Jar)
2. Double click it
3. Add your tasks
4. Let it manage your tasks for you 🔥💯 
And it's **FREE!**

## Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

licongshen12 committed:
Level 0

licongshen12 committed:
Level 1

licongshen12 committed:
Level 2

licongshen12 committed:
Level 3

licongshen12 committed:
Level 4

licongshen12 committed:
Automated Text Ui Testing

licongshen12 committed:
Level 5

licongshen12 committed:
Level 6

licongshen12 committed:
added enums

licongshen12 committed:
Add save functionaliy

licongshen12 committed:
add date and time functionality

licongshen12 committed:
Abstract Ui class

licongshen12 committed:
Extract Storage class

licongshen12 committed:
Extract Parser class and CommandType enum

licongshen12 committed:
Extract Parser class and CommandType enum

licongshen12 committed:
Extract TaskList class

licongshen12 committed:
Merge branch 'branch-A-MoreOOP'

licongshen12 committed:
Add packages

licongshen12 committed:
Merge branch 'add-gradle-support'

licongshen12 committed:
Add JUnit tests

licongshen12 committed:
Add Jar

licongshen12 committed:
Add Javadoc

licongshen12 committed:
Merge branch 'branch-A-JavaDoc'

licongshen12 committed:
Add Level 9

licongshen12 committed:
Merge Level 9

licongshen12 committed:
Fix CheckStyle

licongshen12 committed:
Merge branch 'branch-A-CheckStyle'

licongshen12 committed:
Add JavaFX

licongshen12 committed:
Add JavaFX

licongshen12 committed:
Merge branch-Level-10

licongshen12 committed:
Add Varargs

licongshen12 committed:
Merge branch-A-Varargs

limjunxian1 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,158 @@
+        this.isRunning = false;
+    }
+
+    /**
```
I like how detailed you are in explaining what the code for the getResponse method you have written does, providing me with a better understanding of your implementation and easily compare if your implementation is correct. I noticed you also practice this in the other methods you have implemented which is impressive!

limjunxian1 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,92 @@
+    public static boolean isWithinEventDate(Event event, LocalDate date) {
+        LocalDate startDate = event.getFrom().toLocalDate();
+        LocalDate endDate = event.getTo().toLocalDate();
+        return (date.isEqual(startDate) || date.isEqual(endDate) || (
```
Should this return statement be broken down into a few steps? It seems to me that this return statement is rather complicated and difficult to comprehend at a glance.

limjunxian1 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,119 @@
+     * </p>
+     */
+
+    private void ensureDirectoryExists() {
```
I like how you name your methods, which is consistent throughout across the classes you have created. It clearly explained the named entity accurately and at a sufficient level of detail.

limjunxian1 commented on src/test/java/duke/task/EventTest.java:
```diff
@@ -0,0 +1,32 @@
+                    "E | 0 | test | 16/10/2001 1200 | 20/2/2003 1200",
+                    "'toFileString()' should return a simplified string to be saved to file");
+        } catch (DukeException e) {
+            // do nothing
```
Is there a reason why this catch block is empty? I do notice this in your another test method.

limjunxian1 reviewed (commented):
Overall, I found your code easy to read because of your detailed Java documentation with appropriate naming of your methods and fields. LGTM!

tanveersingh10 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,158 @@
+ */
+public class Duke {
+    /** Task list to store and manage all tasks. */
+    private TaskList taskList;
```
I like how you put all ur variables as private to ensure encapsulation

tanveersingh10 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,95 @@
+     *
+     * @param tasks The list of tasks to initialize with.
+     */
+    public TaskList(Task... tasks) {
```
good use of varargs!

tanveersingh10 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,92 @@
+     * @param input The user input to be parsed.
+     * @return The corresponding {@link CommandType} based on the user input.
+     */
+    public static CommandType parseCommand(String input) {
```
maybe can use switch / case syntax for cleaner code?

tanveersingh10 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,92 @@
+     * @param date  The date to verify.
+     * @return {@code true} if the date is within the event's date range, otherwise {@code false}.
+     */
+    public static boolean isWithinEventDate(Event event, LocalDate date) {
```
i like how the name suggests it returns a boolean!

tanveersingh10 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,208 @@
+     * @param task The task that was marked as done.
+     * @return A string containing the message.
+     */
+    public String displayMarkedAsDone(Task task) {
```
naming of variable is great and its very clear what it does

tanveersingh10 reviewed (commented):
LGTM!

licongshen12 committed:
Update build.gradle

licongshen12 committed:
Add assertions

licongshen12 committed:
Merge branch 'branch-A-Assertions'

licongshen12 committed:
Create methods for handling commandTypes in getResponse()

In the switch statements for getResponse(), the input handling and
processing for each case was done within the switch statements.

Creating new methods to handle each case improves code quality.

licongshen12 committed:
Merge pull request #1 from licongshen12/branch-A-CodeQuality

Create methods for handling commandTypes in getResponse()

licongshen12 committed:
Add gradle.yml

licongshen12 committed:
Add gradle.yml

licongshen12 committed:
Add functionality for sorting deadlines

Add SORT to commandType.java, add getSortedDeadlines() to TaskList.java
so that users can view all the deadlines sorted by earliest deadline
to latest deadline.

licongshen12 committed:
Merge pull request #2 from licongshen12/branch-BCD-Extension

Add functionality for sorting deadlines

licongshen12 committed:
Update README.md

licongshen12 committed:
Add Ui.png
